packages/web-app-preview/src/components/Sources/MediaVideo.vue

```diff
@@ -18,6 +25,23 @@ export default defineComponent({
       type: Boolean,
       default: true
     }
+  },
+  beforeUnmount() {
+    window.removeEventListener('resize', this.resizeVideoDimensions)
+  },
+  methods: {
+    updateDimensions() {
+      this.resizeVideoDimensions()
+      window.addEventListener('resize', this.resizeVideoDimensions)
```

Comment:
Each time this method gets called you register an event listener, is this done intentionally? Wouldn't it make more sense to register the event listener once on mount?

Reply:
Well this method is called only once for each video. It's triggered on loadedmetadata, which is triggered only once and it ensures that videoWidth and videoHeight are available (loadeddata runs before that and at that point those data are not available yet).

Reply:
This condition might be misleading. The height, width and video have to be checked for not being nullable because when you go to next video it triggers, but it's already nullable. So it's here mostly because of file navigation.
```
 if (maxHeight && maxWidth && this.$refs.video) {
```